### ISSUE TITLE
Message list preview fails when API endpoint tries to work with an 'empty' body

### ISSUE THREAD

**Issue by Blitheness** [vanilla/vanilla#11033]
"Please search existing issues to avoid creating duplicates." - Quick search revealed no similar report.

**Describe the bug**
- Conversation happening between multiple members
- Someone sends a message containing only an embed (image)
- Conversations list fails to load when message icon in top-right is clicked

**Vanilla info**
1. Are you sure this is a core Vanilla problem and not caused by any addon you are using?
   - No third-party themes or plugins installed.
2. Did you follow our generic troubleshooting steps already?
   - No. See below.
3. What Vanilla version are you using?
   - RC2 2021.003
4. When did the issue start?
   - When the message list preview box started making an API call that uses vanilla/garden-schema to validate the response

**To Reproduce**
1. Send a conversation message where a single image is embedded, with no other content in the message.
   - Example message content to manually enter into the database to force this issue to appear: `[{"insert":{"embed-external":{"data":{"url":"https:\/\/vanillaforums.com\/images\/casestudies\/hootsuite.jpg","name":"image.png","type":"image\/jpg","size":7066,"width":242,"height":148,"mediaID":123,"dateInserted":"2020-12-16T17:52:50+00:00","insertUserID":2,"foreignType":"embed","foreignID":2,"embedType":"image","format":null,"bodyRaw":null},"loaderData":{"type":"image"}}}},{"insert":"\n"}]`
2. Click on the messages icon in the top-right of the forum which should bring up a pop-out box containing your most recent conversations
3. Observe that it fails to load (loading spinner keeps going), and error 422 appears in console.
   - It tries to call `/api/v2/conversations?expand=all`
4. Change the message content so at the end there is a `.` before `\n`, so `{"insert":".\n"}`, then observe the list loads normally because it thinks the message has content (a single dot/period) and is valid.

**Expected behavior**
I expect the message summary to actually load when I click the button to display it. Vanilla 3.x would interpret the message body as a link to the embedded image, so you'd see something like this:
<img width="306" alt="image" src="https://user-images.githubusercontent.com/19779808/111072757-dca99800-84d3-11eb-8e20-2efa60055207.png">

**Screenshots**
<img width="377" alt="image" src="https://user-images.githubusercontent.com/19779808/111072660-69a02180-84d3-11eb-9f91-9b4a1e4a197a.png">

**Additional context**
The error seems to be happening in "MessagesApiController.php", around like 362 where a validator is invoked:
https://github.com/vanilla/vanilla/blob/master/applications/conversations/controllers/api/MessagesApiController.php

At some point prior to that, it's as if the post body has been trimmed to the point where it is validated as an empty string.

```
{
    "message": "Validation Failed",
    "status": 422,
    "errors": [
        {
            "field": "body",
            "code": "missingField",
            "path": "[1]",
            "status": 422,
            "message": "item[1].body is required."
        }
    ]
}
```

**Comment by Blitheness** [comment]
Hopefully this is fixed in release 2021.008.

"Fix conversations api when body has an external embed"

I'll give it a go.

**Comment by Blitheness** [comment]
Resolved in 8502664b5ecf67348a0a13c5f4f4996fa038d73d
